### ISSUE TITLE
Encoder-decoder architecture

### ISSUE THREAD

**Issue by astrogilda** [ptuls/tabnet-modified#2]
Hey Paul,

Thanks for this repository! Could you help me understand how one would modify the code to function in an encoder decoder capacity, as referenced in the paper?

**Comment by ptuls** [comment]
Hey @astrogilda,

You'll need to create a new `decoder` function, similar to the `encoder` function
```
def decoder(self, data, reuse, is_training)
```
You'll have to implement a new loss function as per the paper for self-supervision
![image](https://user-images.githubusercontent.com/1995167/88501001-843d8d00-d00d-11ea-818f-a22ed34c6c9d.png)
in Section 4.

Then, in [here](https://github.com/ptuls/tabnet-modified/blob/master/train_classifier.py#L121), add a decoder after the encoder to train I think. You might want to remove the classification part too.

At least that's the rough idea I have. Hopefully that helps.
